### ISSUE TITLE
Memory leak on .NET Core

### ISSUE THREAD

**Issue by malmgren80** [aspnet/Microsoft.Data.Sqlite#379]
I'm suspecting a memory leak when running `Microsoft.Data.Sqlite 1.1.1` with .NET Core (tested on runtime 1.1.0 on Windows 10). 

I've made a small program that runs INSERT, UPDATE and SELECT in a while(true) loop. 5 threads is running the loop simultaneously and it eats memory quite quickly - ~25MB -> 300MB memory in about 5 mins. Also tested briefly on net461 where it goes from 15MB -> 25MB in about 5mins (seems slower). I've got a feeling that it eats quicker when larger strings (700-800 bytes) are being inserted/updated/selected.

I have profiled it with dotMemory from JetBrains but it doesnt get a good overview for unmanaged memory.

Connectionstring that is being used is `"Data Source=./data/storage.db"`

The ADO.NET code have the following pattern:
```
private IDbConnection GetOpenConnection()
{
    var connection = new SqliteConnection(ConnectionString);
    connection.Open();
    return connection;
}

public IEnumerable<Foo> GetFoos()
{
    using (var connection = GetOpenConnection())
    using (var command = connection.CreateCommand())
    {
        command.CommandText = @"some sql with params @p1 and @p2";

        var parameter = command.CreateParameter();
        parameter.ParameterName = "@p1";
        parameter.Value = p1 ?? DBNull.Value;
        command.Parameters.Add(parameter);

        var parameter = command.CreateParameter();
        parameter.ParameterName = "@p2";
        parameter.Value = p2 ?? DBNull.Value;
        command.Parameters.Add(parameter);

        using (var reader = command.ExecuteReader())
        {
            while (reader.Read())
            {
                yield return new Foo
                {
                    Bar = reader.GetGuid(0),
                    Baz = reader.GetString(1),
                    FooBar = reader.GetDateTime(2),
                };
            }
        }
    }
}
```

**Comment by ajcvickers** [comment]
@malmgren80 Could you try this with the 2.0 preview2 release? The native interop was changed between the two releases, so it might be that this is fixed with 2.0 already.

**Comment by bricelam** [comment]
I tested this on the nightly builds, and the private memory never went over 6MB.

**Comment by yevgen1scr** [comment]
I've experienced recently same issue while running load test on WebAPI with SQLite at the back-end. 
Everything was running in docker container microsoft/aspnetcore:2.0 (.net core runtime: 2.0.5).
In fact load test was hardly doing much of load - 200 requests in 5 sec (2 threads), every request was doing simple SELECT operation (returning 5 records). 
After the test process consumed +200MB (I was monitoring it in real time on server).
When I increased test load to 10k requests it burned around 4GB of RAM.
